templates/blog/post_detail.html

```diff
@@ -0,0 +1,28 @@
+{% extends "base.html" %}
+
+{% block title %}<title>Post {{ post.id }} details</title>{% endblock %}
+
+{% block content %}
+  <h1>{{ post.title }}</h1>
+  <h6 class="owner_created">{{post.owner}} at {{post.created_time}}, comments: {{ post.comments.count }}</h6>
+  <p>{{post.content}}</p>
+  <br>
+  <h6>{{ post.comments.count }} comment{{ post.comments.count|pluralize }}</h6>
+  {% for comment in post.comments.all.select_related %}
```

Comment:
Why do you use `select_related` here?
```suggestion
  {% for comment in post.comments.all %}
```

Reply:
We have N+1 problem without select_related in current db request.

In the view instance we request posts with prefetch_related for comments and

In the template there is the model-object: post. When we request all data from post.comments we also need select_related for comment's users.

You can see the difference in request number with select_related and without on screens
![Знімок екрана 2023-12-05 о 21 07 47](https://github.com/mate-academy/py-blog/assets/105721786/087e58a6-8452-4caa-88d5-8aa455d1a2fc)
![Знімок екрана 2023-12-05 о 21 08 20](https://github.com/mate-academy/py-blog/assets/105721786/09f034ac-0bc5-4881-9c42-2b5c4f13150c)
![Знімок екрана 2023-12-05 о 21 09 28](https://github.com/mate-academy/py-blog/assets/105721786/ff7df5e8-08b4-48f3-a239-9f740f767dbf)
![Знімок екрана 2023-12-05 о 21 10 12](https://github.com/mate-academy/py-blog/assets/105721786/f67baa59-f674-410d-8c45-55e0a2bee32c)

Reply:
I'm not sure, that you can call select_related inside template, because template mustn't modify a database query. So, can try to modify `queryset` in your view in two ways.
1. Modify get_queryset method of view
def get_queryset(self):
        comments_with_user = models.Prefetch(
            "comments",
            queryset=Commentary.objects.select_related("user"),
        )
        return Post.objects.all().prefetch_related(comments_with_user)
2. Modify queryset of view
queryset = Post.objects.all().prefetch_related("comments__user")

Reply:
> I'm not sure, that you can call select_related inside template, because template mustn't modify a database query.

But it does work, as well as 2nd method.